src/components/commons/Button/Button.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { ReactNode } from "react";
+import * as S from "./Button.styled";
+
+export interface ButtonPropTypes {
+  size: "xlarge" | "large" | "medium" | "small" | "xsmall";
+  isDisabled?: boolean;
+  varient: "primary" | "line" | "gray" | "blue";
+  children: ReactNode; //자식 노드(텍스트)를 받기 위해 필수.
+  onClick: React.MouseEventHandler<HTMLButtonElement> | undefined; //MouseEvent와는 상이
```

Comment:
p3) 여기서도 `?` 로 optional chaining 을 사용하면 굳이 undefined랑 or 연산을 할 필요가 없어 보여요!
관련해서 링크 첨부하겠습니다 ;)

https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Operators/Optional_chaining

```suggestion
  onClick?: React.MouseEventHandler<HTMLButtonElement>
```

Reply:
오 ? 연산자는 널리쉬 값일 경우 알아서 undefined로 바꿔주는거였네요..! 덕분에 코드 더 깔끔해진 것 같습니다 감사합니다 !!